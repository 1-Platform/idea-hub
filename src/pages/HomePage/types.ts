/* eslint-disable @typescript-eslint/no-explicit-any */
import { TagDoc } from 'pouchDB/types';

export interface CreateNewIdea {
  title: string;
  description: string;
  tags: {
    name: string;
  }[];
}

export interface TabType {
  tabIndex: number;
  tabName: 'recent' | 'popular';
}

export interface Filter {
  author: string | null;
  category: string;
}

export interface TagCount {
  isLoading: boolean;
  data: Array<{
    id: any;
    key: any;
    value: any;
    doc?: PouchDB.Core.ExistingDocument<TagDoc & PouchDB.Core.AllDocsMeta>;
  }>;
}
