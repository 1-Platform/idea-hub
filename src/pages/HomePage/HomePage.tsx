export const HomePage = (): JSX.Element => {
    return <h1>Home Page</h1>;
};
