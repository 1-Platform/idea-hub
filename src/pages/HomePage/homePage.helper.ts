import { IdeaModel } from 'pouchDB/api/idea';
import { IdeaDoc } from 'pouchDB/types';
import { binarySearchOfPouchDbDocs } from 'utils/binarySearch';

export const onIdeaChange = async (
  doc: PouchDB.Core.ExistingDocument<IdeaDoc & PouchDB.Core.AllDocsMeta>,
  ideaList: PouchDB.Core.ExistingDocument<IdeaDoc & PouchDB.Core.AllDocsMeta>[],
  idea: IdeaModel
): Promise<PouchDB.Core.ExistingDocument<IdeaDoc & PouchDB.Core.AllDocsMeta>[]> => {
  const newIdeaList = [...ideaList];
  const changedIdeaPos = binarySearchOfPouchDbDocs(newIdeaList, doc._id);
  const formatedDoc = await idea.formatIdea(doc);
  if (
    changedIdeaPos === -1 &&
    (newIdeaList.length === 0 || doc._id.localeCompare(newIdeaList?.[0]?._id) === 1)
  ) {
    newIdeaList.unshift(formatedDoc);
  } else if (changedIdeaPos !== -1) {
    newIdeaList[changedIdeaPos] = formatedDoc;
  }
  return newIdeaList;
};
