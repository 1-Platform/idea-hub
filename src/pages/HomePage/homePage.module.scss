.sticky {
  &-title {
    position: sticky;
    top: 0;
    padding-top: 16px;
    background-color: white;
    z-index: 10;
  }

  &-sidebar {
    position: sticky;
    top: 80px;
    background: white;
    height: max-content;
  }

  &-topbar {
    // this style is applied on menu-bar for making the cards border invisible on scroll
    margin-left: -1.5px;
    width: calc(100% + 3px);
  }
}
