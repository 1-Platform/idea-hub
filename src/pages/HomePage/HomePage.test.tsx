import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ShallowRenderer from 'react-test-renderer/shallow';

import { HomePage } from './HomePage';

afterEach(() => {
  cleanup();
});

test('HomePage matches the snapshot saved', function () {
  const renderer = ShallowRenderer.createRenderer();
  const tree = renderer.render(<HomePage />);
  expect(tree).toMatchSnapshot();
});
