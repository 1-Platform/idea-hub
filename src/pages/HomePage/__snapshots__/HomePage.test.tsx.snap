// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches the snapshot saved 1`] = `
<React.Fragment>
  <Grid
    hasGutter={true}
  >
    <GridItem
      span={3}
    >
      <Flex
        className="sticky-title"
        direction={
          Object {
            "default": "column",
          }
        }
        flexWrap={
          Object {
            "default": "nowrap",
          }
        }
      >
        <FlexItem>
          <Button
            isBlock={true}
            onClick={[Function]}
            variant="primary"
          >
            Post a new idea!
          </Button>
        </FlexItem>
        <FlexItem>
          <Categories />
        </FlexItem>
      </Flex>
    </GridItem>
    <GridItem
      span={9}
    >
      <Flex
        direction={
          Object {
            "default": "column",
          }
        }
        flexWrap={
          Object {
            "default": "nowrap",
          }
        }
      >
        <FlexItem
          className="sticky-title"
        >
          <MenuTab />
        </FlexItem>
        <FlexItem>
          <Stack
            hasGutter={true}
          >
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={true}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
            <StackItem
              className="pf-u-mx-xs"
            >
              <IdeaItem
                commentCount={2}
                hasVoted={false}
                postedOn="6/12/2021"
                title="An internal platform for associate run projects and experimentation"
                user="Mayur Deshmukh"
                voteCount={100}
              />
            </StackItem>
          </Stack>
        </FlexItem>
      </Flex>
    </GridItem>
  </Grid>
  <NewIdeaModal
    handleModalClose={[Function]}
    isOpen={false}
  />
</React.Fragment>
`;
